pkgs/by-name/pe/perfect_dark/package.nix

```
@@ -0,0 +1,130 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  SDL2,
+  cmake,
+  libGL,
+  pkg-config,
+  python3,
+  zlib,
+  romID ? "ntsc-final",
+}:
+let
+  roms = [
+    "ntsc-final"
+    "pal-final"
+    "jpn-final"
+  ];
+in
+assert lib.assertOneOf "romID" romID roms;
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "perfect_dark";
+  version = "0-unstable-2025-08-25";
+
+  src = fetchFromGitHub {
+    owner = "fgsfdsfgs";
+    repo = "perfect_dark";
+    rev = "bb4fcffeb5dc382fce4c609897a2e82590d7d709";
+    hash = "sha256-XLmAjwEzz4fPpHuk3IBmhhDfiuudwMTnYgVe6Wcfdsg=";
+  };
+
+  enableParallelBuilding = true;
+
+  # Fails to build if not set:
+  hardeningDisable = [ "format" ];
+  hardeningEnable = [ "pie" ];
+
+  cmakeFlags = [
+    (lib.cmakeFeature "ROMID" romID)
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    python3
+  ];
+
+  buildInputs = [
+    SDL2
+    libGL
+    zlib
+  ];
+
+  postPatch =
+    # The project uses Git to retrieve version informations but our
+    # fetcher deletes the .git directory, so we replace the commands
+    # with the correct data directly.
+    ''
+      substituteInPlace CMakeLists.txt \
+        --replace-fail "git rev-parse --short HEAD" \
+                       "echo ${builtins.substring 0 9 finalAttrs.src.rev}" \
+        --replace-fail "git rev-parse --abbrev-ref HEAD" \
+                       "echo port"
+    ''
+    # Point toward the compiled binary and not the shell wrapper since
+    # the rom auto-detection logic is not needed in this build.
+    + ''
+      substituteInPlace dist/linux/io.github.fgsfdsfgs.perfect_dark.desktop \
+        --replace-fail "Exec=io.github.fgsfdsfgs.perfect_dark.sh" \
+                       "Exec=io.github.fgsfdsfgs.perfect_dark"
+    '';
+
+  preConfigure = ''
+    patchShebangs --build .
```

```suggestion
    patchShebangs --build tools/assetmgr
```

Don't recursively patch all files in source, but only the files we need for building.